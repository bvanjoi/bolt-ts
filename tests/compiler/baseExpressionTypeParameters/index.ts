// From `github.com/microsoft/TypeScript/blob/v5.8.3/tests/cases/compiler/baseExpressionTypeParameters.ts`, Apache-2.0 License

// Repro from #17829

function base<T>() {
  class Base {
      static prop: T;
  }
  return Base;
}

class Gen<T> extends base<T>() {}  // Error, T not in scope
//~^ ERROR: Cannot find name 'T'.
class Spec extends Gen<string> {}

<string>Spec.prop;