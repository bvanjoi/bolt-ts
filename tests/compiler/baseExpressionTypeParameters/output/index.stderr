  × Cannot find name 'T'.
    ╭─[../../tests/compiler/baseExpressionTypeParameters/index.ts:12:27]
 12 │ T
    · ─
    ╰────
